### PR TITLE
Apply `black` to `astropy.visualization`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to `astropy.visualization` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/visualization/interval.py
Comment (WilliamJamieson) [reply astropy/visualization/interval.py:80]: Not that I know of, but I haven't really tried that hard to find an automatic fix.

Comment (saimn) [reply astropy/visualization/interval.py:80]: Yes `--preview` activates some experimental string processing but it's still experimental:
https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#improved-string-processing
It seems a bit risky to use it as the additional changes caused by this option would be lost in the middle of the other changes, maybe we can give it a try after the black PRs have been merged ?

Comment (eerovaher) [reply astropy/visualization/interval.py:80]: It is possible to review changes one commit at a time, so applying `black --preview` in a separate commit after `black` has already been applied should be manageable. And this will not clutter the history because we'll squash every PR down to a single commit anyways.

Comment (saimn) [reply astropy/visualization/interval.py:80]: Sure it is possible but will people do it ? I think it's a bit late to introduce an experimental option. I use this option in small projects but here given the number of changes it would be easy to miss some weird behavior.

Comment (larrybradley) [reply astropy/visualization/interval.py:80]: I wouldn't want to merge this as is.  So should maintainers manually fix these string concatenations?

Comment (WilliamJamieson) [reply astropy/visualization/interval.py:80]: I'll try go through and fix these concatenations and push a commit.

**Comment by WilliamJamieson** [comment]
@larrybradley I just pushed a commit using the `--preview` flag while running `black`. It seems to have fixed many of the strings in question. Since it is in a separate commit it should be fairly easy to tell if that option has caused any wild changes.

File: astropy/visualization/lupton_rgb.py
Comment (astrofrog) [inline astropy/visualization/lupton_rgb.py:169]: I think it's doing this weird formatting because of the long comment - this is probably a case where we should put the comment on a separate line and re-simplify the code?

File: astropy/visualization/lupton_rgb.py
Comment (astrofrog) [inline astropy/visualization/lupton_rgb.py:173]: As above - no point splitting up the line just because of the comment

File: astropy/visualization/lupton_rgb.py
Comment (astrofrog) [inline astropy/visualization/lupton_rgb.py:235]: As above

File: astropy/visualization/lupton_rgb.py
Comment (astrofrog) [inline astropy/visualization/lupton_rgb.py:271]: As above

File: astropy/visualization/lupton_rgb.py
Comment (astrofrog) [inline astropy/visualization/lupton_rgb.py:287]: As above - wonder if we can identify thee cases automatically?

File: astropy/visualization/lupton_rgb.py
Comment (astrofrog) [inline astropy/visualization/lupton_rgb.py:351]: As above

File: astropy/visualization/tests/test_histogram.py
Comment (astrofrog) [inline astropy/visualization/tests/test_histogram.py:79]: This is probably short enough that it doesn't really matter that it is expanded.

File: astropy/visualization/wcsaxes/tests/test_formatter_locator.py
Comment (astrofrog) [inline astropy/visualization/wcsaxes/tests/test_formatter_locator.py:466]: I think we could use 'match=' in the pytest.raises here to improve readability

File: astropy/visualization/wcsaxes/tests/test_formatter_locator.py
Comment (astrofrog) [inline astropy/visualization/wcsaxes/tests/test_formatter_locator.py:480]: As above we should use match= (I know this is not strictly related to the formatting changes but I'm looking at this code for the first time in a while! Same comment below several times but I won't keep repeating.

File: astropy/visualization/wcsaxes/tests/test_misc.py
Comment (astrofrog) [inline astropy/visualization/wcsaxes/tests/test_misc.py:374]: As above - maybe we really need to just go through all the pytest.raises and make sure they use match=! (I am mentioning this because I find the black formatted code here quite unreadable)

**Review by astrofrog — CHANGES_REQUESTED** [review]
Looks good to me overall! A few comments below - but happy to approve once these are addressed. I won't have much time in the next few days so @WilliamJamieson feel free to make the changes if you have time, but otherwise I can try next week.

File: astropy/visualization/wcsaxes/tests/test_formatter_locator.py
Comment (astrofrog) [reply astropy/visualization/wcsaxes/tests/test_formatter_locator.py:466]: To be clear my comments about using match= aren't crucial for merging this PR

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py]: Fixed by 5dded7a.

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py]: Fixed by 5dded7a.

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py]: Fixed by 5dded7a.

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py]: Fixed by 5dded7a.

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py]: Fixed by 5dded7a.

File: astropy/visualization/wcsaxes/tests/test_formatter_locator.py
Comment (WilliamJamieson) [reply astropy/visualization/wcsaxes/tests/test_formatter_locator.py]: I searched for all of these and fixed them as part of  5dded7a.

File: astropy/visualization/wcsaxes/tests/test_formatter_locator.py
Comment (WilliamJamieson) [reply astropy/visualization/wcsaxes/tests/test_formatter_locator.py]: See  5dded7a.

File: astropy/visualization/wcsaxes/tests/test_misc.py
Comment (WilliamJamieson) [reply astropy/visualization/wcsaxes/tests/test_misc.py]: Fixed by 5dded7a.

**Comment by WilliamJamieson** [comment]
@astrofrog I think I fixed most of the issues you called out. In particular all the `pytest.raises` which test the error messages now use the `match` instead of an `assert`.

**Review by astrofrog — APPROVED** [review]
Looks good to me - @larrybradley @Cadair did you want to take a look before we move this to the next stage of string normalization?

File: astropy/visualization/lupton_rgb.py
Comment (larrybradley) [inline astropy/visualization/lupton_rgb.py:173]: To be consistent, should this be indented one more level?

File: astropy/visualization/lupton_rgb.py
Comment (larrybradley) [inline astropy/visualization/lupton_rgb.py:229]: Is this a newly-added comment?

File: astropy/visualization/lupton_rgb.py
Comment (larrybradley) [inline astropy/visualization/lupton_rgb.py:313]: Can we just delete this line?

**Review by larrybradley — COMMENTED** [review]
Looks good overall, but I have a couple minor comments.

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py:173]: It is at the same level as the other nested `np.where` statements because it is acting at that level. So I think this is right.

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py:229]: Oops this is a stray comment I inserted by accident when refactoring. I'll remove it

File: astropy/visualization/lupton_rgb.py
Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py]: Done

File: astropy/visualization/lupton_rgb.py
Comment (larrybradley) [reply astropy/visualization/lupton_rgb.py:173]: sure, but I was assuming those all should be indented too.  If you look at line 175 below, it's indented relative to `np.where`, but above on line 173 it is not.  So should lines, 173-183 all be indented one level?

Comment (WilliamJamieson) [reply astropy/visualization/lupton_rgb.py:173]: I see, It just so happens that `c = ` is 4 characters, which is what `black` does for each indent which is why it lines up  the the `np.where` the others doing have this because there is not the assignment part at the start.

If this part bothers you too much I can add some `fmt: skip` around it and fix it to something that you would prefer.

Comment (larrybradley) [reply astropy/visualization/lupton_rgb.py:173]: no worries.  Thanks for the explanation.

**Review by larrybradley — APPROVED** [review]

**Comment by WilliamJamieson** [comment]
@astrofrog, @Cadair, and @larrybradley if this  PR is ready for string normalization, please move it the the "Ready for string normalization" column in https://github.com/orgs/astropy/projects/10.

**Comment by taldcroft** [comment]
@astrofrog, @Cadair, and @larrybradley - ping! Will you have time to finish this up today?

**Comment by astrofrog** [comment]
Ready for string normalization/squashing!

**Comment by WilliamJamieson** [comment]
@astrofrog, @Cadair, @larrybradley I have normalized the strings. Please move this to the "Ready for Squash" column if there are no remaining changes to make.

**Comment by astrofrog** [comment]
Ready for squashing!

**Pull request merged**
